diff --git a/Makefile b/Makefile
index 0541ca0..cf77601 100644
--- a/Makefile
+++ b/Makefile
@@ -6,6 +6,8 @@ default: all
 
 ### Build Options ###
 
+include rus_build_options.mk
+
 # These options can either be changed by modifying the makefile, or
 # by building with 'make SETTING=value'. 'make clean' may be required.
 
@@ -173,6 +175,9 @@ DUMMY != ./extract_assets.py $(VERSION) >&2 || echo FAIL
 ifeq ($(DUMMY),FAIL)
   $(error Failed to extract assets)
 endif
+
+include rus_build_content.mk
+
 endif
 
 # Make tools if out of date
