diff --git a/Makefile b/Makefile
index cfac1e8..06ff950 100644
--- a/Makefile
+++ b/Makefile
@@ -400,6 +400,8 @@ else ifneq ($(call find-command,mips64-linux-gnu-ld),)
   CROSS := mips64-linux-gnu-
 else ifneq ($(call find-command,mips64-elf-ld),)
   CROSS := mips64-elf-
+else ifneq ($(call find-command,$(CROSS_MIPS)ld),)
+  CROSS := $(CROSS_MIPS)
 else
   $(error Unable to detect a suitable MIPS toolchain installed)
 endif
@@ -429,7 +431,7 @@ TARGET_CFLAGS := -nostdinc -DTARGET_N64 -D_LANGUAGE_C
 CC_CFLAGS := -fno-builtin
 
 # Check code syntax with host compiler
-CC_CHECK := gcc
+CC_CHECK := $(CROSS_GENERAL)gcc
 CC_CHECK_CFLAGS := -fsyntax-only -fsigned-char $(CC_CFLAGS) $(TARGET_CFLAGS) -std=gnu90 -Wall -Wextra -Wno-format-security -Wno-main -DNON_MATCHING -DAVOID_UB $(DEF_INC_CFLAGS)
 
 # C compiler options
@@ -456,10 +458,10 @@ export LANG := C
 
 else # TARGET_N64
 
-AS := as
+AS := $(CROSS_GENERAL)as
 ifneq ($(TARGET_WEB),1)
-  CC := gcc
-  CXX := g++
+  CC := $(CROSS_GENERAL)gcc
+  CXX := $(CROSS_GENERAL)g++
 else
   CC := emcc
 endif
@@ -468,8 +470,8 @@ ifeq ($(CXX_FILES),"")
 else
   LD := $(CXX)
 endif
-OBJDUMP := objdump
-OBJCOPY := objcopy
+OBJDUMP := $(CROSS_GENERAL)objdump
+OBJCOPY := $(CROSS_GENERAL)objcopy
 PYTHON := python3
 
 # Platform-specific compiler and linker flags
@@ -530,7 +532,7 @@ ifneq (,$(call find-command,clang))
   CPP      := clang
   CPPFLAGS := -E -P -x c -Wno-trigraphs $(DEF_INC_CFLAGS)
 else
-  CPP      := cpp
+  CPP      := $(CROSS_GENERAL)cpp
   CPPFLAGS := -P -Wno-trigraphs $(DEF_INC_CFLAGS)
 endif
 
