diff --git a/Makefile b/Makefile
index cfac1e8..3531611 100644
--- a/Makefile
+++ b/Makefile
@@ -29,6 +29,9 @@ TARGET_WEB ?= 0
 COMPILER ?= ido
 $(eval $(call validate-option,COMPILER,ido gcc))
 
+# USE_ANOTHER_MACHINE - uses another machine, e.g. x86_64-linux-gnu
+USE_ANOTHER_MACHINE ?= $(TARGET_N64)
+
 # Automatic settings only for ports
 ifeq ($(TARGET_N64),0)
 
@@ -87,6 +90,16 @@ ifeq ($(COMPILER),gcc)
   NON_MATCHING := 1
 endif
 
+ifeq ($(USE_ANOTHER_MACHINE),1)
+ifneq ($(call find-command,x86_64-linux-gnu-gcc),)
+  CROSS_MACHINE := x86_64-linux-gnu-
+endif
+ifneq ($(call find-command,x86_64-linux-gnu-gcc-13),)
+  CC_VERSION := -13
+  CROSS_MACHINE := x86_64-linux-gnu-
+endif
+endif
+
 # VERSION - selects the version of the game to build
 #   jp - builds the 1996 Japanese version
 #   us - builds the 1996 North American version
@@ -400,13 +413,17 @@ else ifneq ($(call find-command,mips64-linux-gnu-ld),)
   CROSS := mips64-linux-gnu-
 else ifneq ($(call find-command,mips64-elf-ld),)
   CROSS := mips64-elf-
+else ifneq ($(call find-command,mips-suse-linux-ld),)
+  CROSS := mips-suse-linux-
+else ifneq ($(call find-command,$(CROSS_MIPS)ld),)
+  CROSS := $(CROSS_MIPS)
 else
   $(error Unable to detect a suitable MIPS toolchain installed)
 endif
 
 AS        := $(CROSS)as
 ifeq ($(COMPILER),gcc)
-  CC      := $(CROSS)gcc
+  CC      := $(CROSS)gcc$(CC_VERSION)
 else
   ifeq ($(USE_QEMU_IRIX),1)
     IRIX_ROOT := $(TOOLS_DIR)/ido5.3_compiler
@@ -429,7 +446,7 @@ TARGET_CFLAGS := -nostdinc -DTARGET_N64 -D_LANGUAGE_C
 CC_CFLAGS := -fno-builtin
 
 # Check code syntax with host compiler
-CC_CHECK := gcc
+CC_CHECK := $(CROSS_MACHINE)gcc$(CC_VERSION)
 CC_CHECK_CFLAGS := -fsyntax-only -fsigned-char $(CC_CFLAGS) $(TARGET_CFLAGS) -std=gnu90 -Wall -Wextra -Wno-format-security -Wno-main -DNON_MATCHING -DAVOID_UB $(DEF_INC_CFLAGS)
 
 # C compiler options
@@ -456,10 +473,10 @@ export LANG := C
 
 else # TARGET_N64
 
-AS := as
+AS := $(CROSS_MACHINE)as
 ifneq ($(TARGET_WEB),1)
-  CC := gcc
-  CXX := g++
+  CC := $(CROSS_MACHINE)gcc$(CC_VERSION)
+  CXX := $(CROSS_MACHINE)g++$(CC_VERSION)
 else
   CC := emcc
 endif
@@ -468,8 +485,8 @@ ifeq ($(CXX_FILES),"")
 else
   LD := $(CXX)
 endif
-OBJDUMP := objdump
-OBJCOPY := objcopy
+OBJDUMP := $(CROSS_MACHINE)objdump
+OBJCOPY := $(CROSS_MACHINE)objcopy
 PYTHON := python3
 
 # Platform-specific compiler and linker flags
@@ -530,7 +547,7 @@ ifneq (,$(call find-command,clang))
   CPP      := clang
   CPPFLAGS := -E -P -x c -Wno-trigraphs $(DEF_INC_CFLAGS)
 else
-  CPP      := cpp
+  CPP      := $(CROSS_MACHINE)cpp$(CC_VERSION)
   CPPFLAGS := -P -Wno-trigraphs $(DEF_INC_CFLAGS)
 endif
 
diff --git a/tools/Makefile b/tools/Makefile
index 92a3916..f4a1ab8 100644
--- a/tools/Makefile
+++ b/tools/Makefile
@@ -3,8 +3,8 @@ include ../util.mk
 HOST_ENV := $(shell uname 2>/dev/null || echo Unknown)
 HOST_ENV := $(patsubst MINGW%,MinGW,$(HOST_ENV))
 
-CC           := gcc
-CXX          := g++
+CC           := gcc$(CC_VERSION)
+CXX          := g++$(CC_VERSION)
 CFLAGS       := -I . -Wall -Wextra -Wno-unused-parameter -pedantic -O2 -s
 LDFLAGS      := -lm
 ALL_PROGRAMS := armips n64graphics n64graphics_ci mio0 n64cksum textconv patch_elf_32bit aifc_decode aiff_extract_codebook vadpcm_enc tabledesign extract_data_for_mio skyconv
diff --git a/tools/audiofile/Makefile b/tools/audiofile/Makefile
index ae7d1e0..16cb023 100644
--- a/tools/audiofile/Makefile
+++ b/tools/audiofile/Makefile
@@ -1,4 +1,6 @@
-CXX := g++
+include ../../util.mk
+
+CXX := g++$(CC_VERSION)
 
 libaudiofile.a: audiofile.o
 	$(AR) rcs $@ $^
diff --git a/tools/ido5.3_recomp/Makefile b/tools/ido5.3_recomp/Makefile
index 84a5de3..a260ea8 100644
--- a/tools/ido5.3_recomp/Makefile
+++ b/tools/ido5.3_recomp/Makefile
@@ -1,3 +1,8 @@
+include ../../util.mk
+
+CC := gcc$(CC_VERSION)
+CXX := g++$(CC_VERSION)
+
 IRIX_ROOT := ../ido5.3_compiler
 
 cc: OPT_CFLAGS := -O2
diff --git a/util.mk b/util.mk
index 75af981..295625d 100644
--- a/util.mk
+++ b/util.mk
@@ -14,3 +14,8 @@ endef
 
 # Returns the path to the command $(1) if exists. Otherwise returns an empty string.
 find-command = $(shell which $(1) 2>/dev/null)
+
+# Find that GCC version is 13 or is in machine x86_64-linux-gnu-
+ifneq ($(call find-command,gcc-13),)
+  CC_VERSION := -13
+endif
