diff --git a/Makefile b/Makefile
index cfac1e8..5c27de2 100644
--- a/Makefile
+++ b/Makefile
@@ -394,16 +394,31 @@ DEF_INC_CFLAGS := $(foreach i,$(INCLUDE_DIRS),-I$(i)) $(C_DEFINES)
 ifeq ($(TARGET_N64),1)
 
 # detect prefix for MIPS toolchain
-ifneq      ($(call find-command,mips-linux-gnu-ld),)
+ifneq      ($(call find-command,$(MIPS_CROSS)-ld),)
+  CROSS := $(MIPS_CROSS)-
+else ifneq ($(call find-command,mips-linux-gnu-ld),)
   CROSS := mips-linux-gnu-
 else ifneq ($(call find-command,mips64-linux-gnu-ld),)
   CROSS := mips64-linux-gnu-
 else ifneq ($(call find-command,mips64-elf-ld),)
   CROSS := mips64-elf-
+else ifneq ($(call find-command,mips-suse-linux-ld),)
+  CROSS := mips-suse-linux-
 else
   $(error Unable to detect a suitable MIPS toolchain installed)
 endif
 
+# detect GCC for AMD64 or i386 architecture installed
+ifneq      ($(call find-command,$(CC_CHECK_CROSS)-gcc),)
+  CC_CHECK := $(CC_CHECK_CROSS)-gcc
+else ifneq ($(call find-command,x86_64-linux-gnu-gcc),)
+  CC_CHECK := x86_64-linux-gnu-gcc
+else ifneq ($(call find-command,i686-linux-gnu-gcc),)
+  CC_CHECK := i686-linux-gnu-gcc
+else
+  $(error Unable to detect GCC for AMD64 or i386 architecture installed)
+endif
+
 AS        := $(CROSS)as
 ifeq ($(COMPILER),gcc)
   CC      := $(CROSS)gcc
@@ -429,7 +444,6 @@ TARGET_CFLAGS := -nostdinc -DTARGET_N64 -D_LANGUAGE_C
 CC_CFLAGS := -fno-builtin
 
 # Check code syntax with host compiler
-CC_CHECK := gcc
 CC_CHECK_CFLAGS := -fsyntax-only -fsigned-char $(CC_CFLAGS) $(TARGET_CFLAGS) -std=gnu90 -Wall -Wextra -Wno-format-security -Wno-main -DNON_MATCHING -DAVOID_UB $(DEF_INC_CFLAGS)
 
 # C compiler options
