diff --git a/Makefile b/Makefile
index cfac1e8..7de349f 100644
--- a/Makefile
+++ b/Makefile
@@ -446,9 +446,6 @@ RSPASMFLAGS := $(foreach d,$(DEFINES),-definelabel $(subst =, ,$(d)))
 ifeq ($(shell getconf LONG_BIT), 32)
   # Work around memory allocation bug in QEMU
   export QEMU_GUEST_BASE := 1
-else
-  # Ensure that gcc treats the code as 32-bit
-  CC_CHECK_CFLAGS += -m32
 endif
 
 # Prevent a crash with -sopt
